File: 프로그래머스/lv1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for (auto i : commands) {
        int start = i[0]-1;
        int end = i[1]-1;
        int idx = i[2]-1;
        vector<int> v;
        for (int j = start; j<=end; j++) {
            v.push_back(array[j]);
        }
        sort(v.begin(), v.end());
        answer.push_back(v[idx]);
    }
       
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.py
def solution(array, commands):
    answer = []
    #for idx, command in enumerate(commands):
            #answer.append(sorted(array[command[0]-1:command[1]])[command[2]-1])
    
    return list(map(lambda x : sorted(array[x[0]-1:x[1]])[x[2]-1], commands))
    #return answer
    